src/drawing/canvas.rs

```diff
@@ -0,0 +1,65 @@
+use image::{
+    GenericImage,
+    GenericImageView,
+    Pixel
+};
+
+/// A surface for drawing on - many drawing functions in this
+/// library are generic over a `Canvas` to allow the user to
+/// configure e.g. whether to use blending.
+///
+/// All instances of `GenericImage` implement `Canvas`, with
+/// the behaviour of `draw_pixel` being equivalent to calling
+/// `set_pixel` with the same arguments.
+pub trait Canvas {
+    /// The type of `Pixel` that can be drawn on this canvas.
+    type Pixel;
```

Comment:
Is it intentional that this does not require `Pixel: image::Pixel`? The trait bound seems mentioned everywhere `Canvas` as used, so introducing it here would allow omitting it. This happens with the current `GenericImage` trait bound.

Reply:
There's no good reason for the current behaviour. I've pushed a commit to adopt your suggestion.